lib/data.js

```diff
@@ -1,23 +1,54 @@
+const { cloudinary } = require("./cloudinary");
 const { collection } = require("./database");
 
-async function getImageDataByTag(userName) {
-  const imgEntry = await collection("users")
-    .find({
-      userName: userName,
-    })
-    .toArray();
+async function getImageDataOfUser(userName) {
+  const imgEntry = await collection("users").findOne({
+    userName: userName,
+  });
   if (!imgEntry) {
     return null;
   }
   console.log({ imgEntry });
   return imgEntry;
 }
 
-exports.getImageDataByTag = getImageDataByTag;
+async function setImage(image, userName) {
+  const uploadedResponse = await cloudinary.uploader.upload(image, {
+    upload_preset: "Tagtrace",
+  });
+  const imageObj = {
+    url: uploadedResponse.secure_url,
+    imgNr: uploadedResponse.asset_id,
+    timeStamp: uploadedResponse.created_at,
+    width: uploadedResponse.width,
+    height: uploadedResponse.height,
+    tags: uploadedResponse.tags,
+  };
+  return await collection("users").updateOne(
+    { userName: userName },
+    { $push: { images: imageObj } }
+  );
+}
 
-// async function getByID({ collectionName, id }) {
-//   const cursor = collection(collectionName).findOne({
-//     _id: ObjectId(id),
+// async function setImage(imageObj, userName) {
+//   const userData = await collection("users").findOne({
+//     name: userName,
```

Comment:
remove comments, if you don't need them.

Reply:
👍